pkg/operator/ceph/object/objectstore.go

```diff
@@ -88,18 +98,89 @@ func deleteRealmAndPools(context *Context, spec cephv1.ObjectStoreSpec) error {
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone group")
+	}
+	masterZoneID, err := decodeMasterZoneID(zoneGroupOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse master zone id")
+	}
+
+	zoneOutput, err := RunAdminCommandNoRealm(context, "zone", "get", realmArg, zoneGroupArg, zoneArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone")
+	}
+	zoneID, err := decodeID(zoneOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse zone id")
+	}
+
+	if zoneID == masterZoneID {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func getZoneEndpoints(context *Context, realmArg string, zoneGroupArg string, zoneName string, serviceEndpoint string) (string, error) {
+	zoneGroupOutput, err := RunAdminCommandNoRealm(context, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get rgw zone group")
+	}
+	zones, err := decodeZonesList(zoneGroupOutput)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse zones list")
+	}
+
+	numZones := len(zones)
+	zoneEndpoints := ""
+	for i := 0; i < numZones; i++ {
+		if zones[i].Name == zoneName {
+			numEndpoints := len(zones[i].Endpoints)
+			for j := 0; j < numEndpoints; j++ {
+				// in case object-store operator code is rereconciled, zone modify could get run again with serviceEndpoint added again
+				if zones[i].Endpoints[j] != serviceEndpoint {
+					zoneEndpoints += zones[i].Endpoints[j] + ","
+				}
+			}
+			break
+		}
+	}
+
+	zoneEndpoints += serviceEndpoint
+
+	return zoneEndpoints, nil
+}
+
 func setMultisite(context *Context, serviceIP string, spec cephv1.ObjectStoreSpec, realmName string, zoneGroupName string, zoneName string) error {
 	realmArg := fmt.Sprintf("--rgw-realm=%s", realmName)
 	zoneGroupArg := fmt.Sprintf("--rgw-zonegroup=%s", zoneGroupName)
 	zoneArg := fmt.Sprintf("--rgw-zone=%s", zoneName)
-	endpointArg := fmt.Sprintf("--endpoints=%s:%d", serviceIP, spec.Gateway.Port)
 	updatePeriod := false
+	serviceEndpoint := fmt.Sprintf("%s:%d", serviceIP, spec.Gateway.Port)
+
+	zoneEndpoints, err := getZoneEndpoints(context, realmArg, zoneGroupArg, zoneName, serviceEndpoint)
+	if err != nil {
+		return err
+	}
+	logger.Debugf("Endpoints for zone %q are: %q", zoneName, zoneEndpoints)
+	endpointArg := fmt.Sprintf("--endpoints=%s", zoneEndpoints)
 
 	if spec.IsMultisite() {
 		updatePeriod = true
-		_, err := RunAdminCommandNoRealm(context, "zonegroup", "modify", realmArg, zoneGroupArg, endpointArg)
+		zoneIsMaster, err := checkZoneIsMaster(context, realmArg, zoneGroupArg, zoneArg)
 		if err != nil {
-			return errors.Wrapf(err, "failed to add object store %q in rgw zone group %q", context.Name, zoneGroupName)
+			return err
+		}
+
+		if zoneIsMaster {
+			_, err := RunAdminCommandNoRealm(context, "zonegroup", "modify", realmArg, zoneGroupArg, endpointArg)
```

Comment:
What is this doing? Updating the endpoints available in the zonegroup? How about adding a comment?

Reply:
The endpoints that are part of a master zone are supposed to be the endpoints for a zone group. These endpoints are used in the zone group for load balancing purpose.